### PR TITLE
Fix "Reading past RLE/BitPacking stream" when querying array columns from parquets

### PR DESCRIPTION
**Pull request by kgalieva**
ParquetColumnReader.remainingValueCountInPage contains number of array elements, not number of arrays. So when parquet consists of multiple pages, the loop exceeds the number of page elements.

### CODE REVIEW

**Comment by petroav** [comment]
@kgalieva thank you for the contribution! Can you please ensure that your commit message follows the guidelines outlined here: http://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html?

**Comment by raghavsethi** [comment]
@nezihyigitbasi if this fits within your Parquet skillset please take a look, otherwise I can do it.

**Comment by nezihyigitbasi** [comment]
@kgalieva I will take a look at this. Before that can you please add a unit test? Also please update your commit message according to our guidelines: https://chris.beams.io/posts/git-commit/

File: presto-hive/src/main/java/com/facebook/presto/hive/parquet/reader/ParquetColumnReader.java [hunk 1]
```diff
@@ -170,32 +170,41 @@ public Block readPrimitive(Type type, IntList positions)
...
-                try {
-                    valueCount++;
-                    repetitionLevel = repetitionReader.readLevel();
-                    if (repetitionLevel == 0) {
-                        positions.add(valueCount);
-                        valueCount = 0;
-                        if (i == numValues - 1) {
-                            return;
-                        }
+        int rowElementsCount = 0;
```
Comment (nezihyigitbasi) [inline presto-hive/src/main/java/com/facebook/presto/hive/parquet/reader/ParquetColumnReader.java]: Please rename these variables to be more descriptive. Currently it's hard to follow what they do. Also in this part of the code we use many different terms like element, position, value, row. Please also make your use of these words consistent.

IIUC, `rowsCount` refers to the total number of values read so far, and `totalCount` refers to the number of values read in the current page page (something like `valuesReadInCurrentPage`). I am not sure about `rowElementsCount`, what does it mean for primitives?

File: presto-hive/src/main/java/com/facebook/presto/hive/parquet/reader/ParquetColumnReader.java [hunk 2]
```diff
@@ -231,6 +240,7 @@ private void seek()
         while (valuePosition < readOffset) {
             if (page == null) {
                 readNextPage();
+                repetitionLevel = repetitionReader.readLevel();
```
Comment (nezihyigitbasi) [inline presto-hive/src/main/java/com/facebook/presto/hive/parquet/reader/ParquetColumnReader.java]: why are you calling `readLevel()` here?

**Comment by renato2099** [comment]
quick question @kgalieva @nezihyigitbasi, is this going into presto 0.188? any idea when is this going to be out? many thanks in advanced!

**Comment by nezihyigitbasi** [comment]
No this won't go into 0.188. @kgalieva has another patch that I am reviewing right now, which may fix this issue.

**Comment by renato2099** [comment]
thanks for the quick reply @nezihyigitbasi ! This is sad though :( I mean that this doesn't fix the problem and that there is still another problem related to this.
Do you have an issue# where I could follow/help on this up please? thanks again!

**Comment by nezihyigitbasi** [comment]
There are multiple Parquet PRs out there awaiting review:
- #9156
- #9205
- #9110
